src/EnergyPlus/ExtendedHI.cc

```diff
@@ -0,0 +1,501 @@
+            if (flux3 < 0.0) {
+                varname = static_cast<int>(EqvarName::Rs);
+                General::SolveRoot(
+                    state,
+                    tol,
+                    maxIter,
+                    SolFla,
+                    Ts,
+                    [&](Real64 Ts) { return (Ts - Ta) / Ra_un(Ts, Ta) + (Pc - Pa) / (Zs((Tc - Ts) / (Q - Qv(Ta, Pa))) + Za_un) - (Q - Qv(Ta, Pa)); },
+                    0.0,
+                    Tc);
+                Rs = (Tc - Ts) / (Q - Qv(Ta, Pa));
+                ZsRs = Zs(Rs);
+                Real64 Ps = Pc - (Pc - Pa) * ZsRs / (ZsRs + Za_un);
+                if (Ps > phi_salt * pvstar(Ts)) {
+                    General::SolveRoot(
+                        state,
+                        tol,
+                        maxIter,
+                        SolFla,
+                        Ts,
+                        [&](Real64 Ts) { return (Ts - Ta) / Ra_un(Ts, Ta) + (phi_salt * pvstar(Ts) - Pa) / Za_un - (Q - Qv(Ta, Pa)); },
+                        0.0,
+                        Tc);
+                    Rs = (Tc - Ts) / (Q - Qv(Ta, Pa));
+                }
+                return Rs;
+            } else {
+                varname = static_cast<int>(EqvarName::DTcdt);
+                Rs = 0.0;
+                dTcdt = (1.0 / C) * flux3;
+                return dTcdt;
+            }
+        }
+    }
+
+    // Convert the find_T function
+    Real64 find_T(EnergyPlusData &state, int eqvar_name, Real64 eqvar)
+    {
+        Real64 T;
+        int SolFla;
+
+        if (eqvar_name == static_cast<int>(EqvarName::Phi)) {
+            General::SolveRoot(
+                state, tol, maxIter, SolFla, T, [&](Real64 T) { return find_eqvar_phi(state, T, 1.0) - eqvar; }, 0.0, 240.0);
+        } else if (eqvar_name == static_cast<int>(EqvarName::Rf)) {
+            General::SolveRoot(
+                state,
+                tol,
+                maxIter,
+                SolFla,
+                T,
+                [&](Real64 T) { return (find_eqvar_Rf(state, T, std::min(1.0, Pa0 / pvstar(T)))) - eqvar; },
+                230.0,
+                300.0);
+        } else if (eqvar_name == static_cast<int>(EqvarName::Rs)) {
+            General::SolveRoot(
+                state, tol, maxIter, SolFla, T, [&](Real64 T) { return find_eqvar_rs(state, T, Pa0 / pvstar(T)) - eqvar; }, 295.0, 350.0);
+        } else {
+            General::SolveRoot(
+                state, tol, maxIter, SolFla, T, [&](Real64 T) { return find_eqvar_dTcdt(state, T, Pa0 / pvstar(T)) - eqvar; }, 340.0, 1000.0);
+        }
+
+        return T;
+    }
+
+    Real64 heatindex(EnergyPlusData &state, Real64 Ta, Real64 RH)
+    {
+
+        // Ta: temperature in Kelvin
+        // RH: relative humidity in range of 0.0 to 1.0
+        // The function computes the extended heat index, in Kelvinn
+
+        auto const HVACSystemRootSolverBackup = state.dataRootFinder->HVACSystemRootFinding.HVACSystemRootSolver;
```

Comment:
I know this caught my attention before because I was thinking you were making a copy of a whole solver object.  It would be great to rename this to like "RootSolverMethod" or something to make clear it's just a flag.

Reply:
yeah, it does sounds like a large object. I just renamed it to HVACSystemRootSolverMethod